plugins/de/opendiabetes/vault/plugin/importer/crawler/CommandLineArgumentParser.java

```diff
@@ -0,0 +1,469 @@
+     * The date until when the csv file data should go.
+     */
+    private String toDate = null;
+
+    /**
+     * This counter will check if it is
+     * 0 then at very last throw a error
+     * or else different modules have
+     * already handled error with
+     * appropriate message.
+     */
+    private int tempCounterForThrowingError = 0;
+
+    /**
+     * The number of iterations for the hashing algorithm.
+     *
+     * Decreasing this speeds down startup time and
+     * can be useful during testing,
+     * but it also makes it easier for brute force attackers.
+     */
+    private static final int ITERATION_COUNT = 40000;
+
+    /**
+     * The length of the key.
+     * Other values give me java.security.InvalidKeyException: Illegal key
+     * size or default parameters.
+     */
+    private static final int KEY_LENGTH = 128;
+
+    /**
+     * Will run program depending on flag/argument chosen.
+     *
+     * @param args - the arguments for the application.
+     * @param logger - a logger instance.
+     * @throws IOException - thrown if any input/output failure occurred.
+     * @throws ParseException - thrown if there was an error parsing the arguments.
+     * @throws SecurityException - thrown if there was a security error.
+     * @throws AWTException - thrown if there was a awt error.
+     * @throws InterruptedException - thrown if there was a threading error.
+     * @throws java.text.ParseException - thrown if there was an error parsing the arguments.
+     * @throws GeneralSecurityException - thrown if there was a security error.
+     */
+    public void runDifferentArguments(final String[] args, final Logger logger)
+            throws IOException, ParseException, SecurityException,
+            AWTException, InterruptedException, java.text.ParseException, GeneralSecurityException {
+
+        logger.info("Inside Class FlagArgumentsClass");
+        CommandLine commandLine; // to get command line arguments
+        if (!(args.length > 0)) {
+            logger.info("Arguments missing");
+            System.out.println("Arguments missing");
+            return;
+        }
+
+        // If there are any argument provided, program will go furthure and
+        // check different modules
+			/*
+			 * If first argument is for help, all other arguments will be
+			 * ignored
+			 */
+
+        Options optionsForHelp = new Options();
+
+        Option shortHelpOptions = Option.builder("h").required(false).desc("The short help option").longOpt("help").build();
+        optionsForHelp.addOption((org.apache.commons.cli.Option) shortHelpOptions);
+
+        commandLine = new DefaultParser().parse(optionsForHelp, args, true);
+        if (commandLine.getOptions().length != 0) {
+            if (commandLine.getOptions()[0].getOpt() == "h") {
+                System.out.println("Version: Carelink Crawler v0.1\n" + "Options: "
+                        + "	  -v,--version       	show program's version number and exit\n"
```

Comment:
Wäre doch schöner, wenn in der Ausgabe alle Optionen auf der Selben höhe stünden?

Reply:
das wird als Liste untereinander angezeigt, wie bei anderen cli programmen eig.

```
Options:
  -v, --version              print Node.js version
  -e, --eval script          evaluate script
  -p, --print                evaluate script and print result
```